src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509Encoder.cs

```diff
@@ -174,7 +174,10 @@ public unsafe void DecodeX509KeyUsageExtension(byte[] encoded, out X509KeyUsageF
                 if (data->pathlen != IntPtr.Zero)
                 {
                     hasPathLengthConstraint = true;
-                    pathLengthConstraint = Interop.libcrypto.ASN1_INTEGER_get(data->pathlen).ToInt32();
+                    long pathLengthConstraint64 = Interop.Crypto.Asn1IntegerGet(data->pathlen);
+
+                    Debug.Assert(pathLengthConstraint64 < int.MaxValue, "pathLengthConstraint needs to be in the Int32 range.");
```

Comment:
Should the shim return an int32_t then?  Or should this be an exception if it's possible it actually could be larger than  Int32.MaxValue?